pkg/ebpf/events_pipeline.go

```diff
@@ -241,14 +243,69 @@ func (t *Tracee) decodeEvents(outerCtx context.Context, sourceChan chan []byte)
 	return out, errc
 }
 
-// shouldProcessEvent decides whether or not to drop an event before further processing it
-func (t *Tracee) shouldProcessEvent(event trace.Event) bool {
+// computeScopes iterates through the scopes that do the filtering in user space, checking whether an event should be considered.
+// If it should not, it sets the respective offset to 0.
+// Finally it returns the bitmask of computed scopes.
+func (t *Tracee) computeScopes(event *trace.Event) uint64 {
 	eventID := events.ID(event.EventID)
+	origMatchedScopes := event.MatchedScopes
+	matchedScopes := event.MatchedScopes
+
+	for filterScope := range t.config.FilterScopes.Map() {
+		bitOffset := uint(filterScope.ID)
+
+		// Events submitted with matching scopes.
+		// The scope must have its bit cleared when it does not match.
+		if utils.HasBit(origMatchedScopes, bitOffset) {
+			// TODO: check if the event is traceable to some scope using a BPF map
+			if !filterScope.IsEventTraceable(eventID) {
+
+				utils.ClearBit(&matchedScopes, bitOffset)
+				continue
+			}
+
+			if !filterScope.ContextFilter.Filter(*event) {
+				utils.ClearBit(&matchedScopes, bitOffset)
+				continue
+			}
+
+			if !filterScope.RetFilter.Filter(eventID, int64(event.ReturnValue)) {
+				utils.ClearBit(&matchedScopes, bitOffset)
+				continue
+			}
+
+			if !filterScope.ArgFilter.Filter(eventID, event.Args) {
+				utils.ClearBit(&matchedScopes, bitOffset)
+				continue
+			}
+
```

Comment:
could you add a comment explaining why you're running UID and PID filter in here as well now ? Since last shouldProcessEvent did not run them.